room/compiler-processing/src/main/java/androidx/room/compiler/processing/JavaPoetExt.kt

```diff
@@ -51,12 +51,9 @@ internal fun TypeMirror.safeTypeName(): TypeName = if (kind == TypeKind.NONE) {
  * Adds the given element as the originating element for compilation.
  * see [TypeSpec.Builder.addOriginatingElement].
  */
-fun TypeSpec.Builder.addOriginatingElement(element: XElement) {
-    if (element is JavacElement) {
```

Comment:
Pulled this implementation out to share with the kotlin poet extension, and to make it public in case any one wants their own custom file writing.

I also changed this to return the builder, which is the pattern that javapoet uses and allows better chaining